packages/babel-helper-create-regexp-features-plugin/src/util.js

```diff
@@ -0,0 +1,48 @@
+import { FEATURES, hasFeature } from "./features";
+
+export function generateRegexpuOptions(node, features) {
+  let useUnicodeFlag = false,
+    dotAllFlag = false,
+    unicodePropertyEscape = false,
+    namedGroup = false;
+  const { flags, pattern } = node;
+  const flagsIncludesU = flags.includes("u");
+
+  if (flagsIncludesU === true) {
+    if (hasFeature(features, FEATURES.unicodeFlag) === false) {
+      useUnicodeFlag = true;
+    }
+    if (
+      hasFeature(features, FEATURES.unicodePropertyEscape) === true &&
+      /\\[pP]{/.test(pattern) === true
+    ) {
+      unicodePropertyEscape = true;
+    }
+  }
+
+  if (hasFeature(features, FEATURES.dotAllFlag) && flags.indexOf("s") >= 0) {
+    dotAllFlag = true;
+  }
+  if (
+    hasFeature(features, FEATURES.namedCaptureGroups) &&
+    /\(\?<(?![=!])/.test(pattern)
+  ) {
+    namedGroup = true;
+  }
+  if (
+    namedGroup === false &&
+    unicodePropertyEscape === false &&
+    dotAllFlag === false &&
+    !(flagsIncludesU === true && hasFeature(features, FEATURES.unicodeFlag))
```

Comment:
Nit: we could rewrite the last line as `(!flagsIncludesU || useUnicodeFlag)` and avoid the function call.

Reply:
In practice, the function call will be inlined since it is so small. But your suggested conditions are easier to understand.